### PR TITLE
Extract element from singleton set

### PR DESCRIPTION
**Pull request by stefan-aws**
By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

**Review by davidcok — CHANGES_REQUESTED** [review]
Should have  an example that uses it -- in particular one that shows why this method should be used over simply `x :| x in s`

File: src/Collections/Sets/Sets.dfy [hunk 1]
```diff
@@ -76,6 +76,36 @@ module {:options "-functionSyntax:4"} Sets {
...
+    }
+    if IsSingleton(s) {
+      var x :| x in s;
+      assert s == {x};
+      assert |s| == 1;
+    }
+  }
+
+  /* Extracts the unique element from a singleton set. */
+  method ExtractFromSingleton<T>(s: set<T>) returns (x: T)
```
[the remark this answers is not in the document]
Comment (stefan-aws) [reply src/Collections/Sets/Sets.dfy]: Changed it to `function` and added context

File: src/Collections/Sets/Sets.dfy
Comment (atomb) [inline src/Collections/Sets/Sets.dfy:1]: What's the long-term plan for these parallel trees? It seems like they'll be hard to maintain.

**Review by atomb — APPROVED** [review]

File: src/Collections/Sets/Sets.dfy
Comment (stefan-aws) [reply src/Collections/Sets/Sets.dfy:1]: My understanding is that as we move forward, new code to the library should only be added to the `dafny` subfolder.

**Pull request merged**